sympde/expr/evaluation.py

```diff
@@ -1140,7 +1140,7 @@ def eval(cls, *_args, **kwargs):
 
             return ImmutableDenseMatrix(lines)
 
-        elif isinstance(expr, DomainExpression):
+        elif isinstance(expr, DomainExpression) or isinstance(expr, BoundaryExpression):
```

Comment:
The intrinsic function `isinstance` accepts a tuple of types as second argument:
```suggestion
        elif isinstance(expr, (DomainExpression, BoundaryExpression)):
```

Reply:
28a2cc124c130a41930b5c786ab9d5ef3b610aed